3.4: broken PowerPC build

[Build](http://pullrequest.opencv.org/buildbot/builders/3_4_powerpc-64le-lin64/builds/800)
```
In file included from /build/3_4_powerpc-64le-lin64/opencv/modules/features2d/src/sift.dispatch.cpp:76:
/build/3_4_powerpc-64le-lin64/opencv/modules/features2d/src/sift.simd.hpp: In function 'void cv::cpu_baseline::calcSIFTDescriptor(const cv::Mat&, cv::Point2f, float, float, int, int, float*)':
/build/3_4_powerpc-64le-lin64/opencv/modules/features2d/src/sift.simd.hpp:659:65: error: no matching function for call to 'v_fma(cv::hal_baseline::v_int32x4, const v_int32&, cv::hal_baseline::v_int32x4)'
             v_int32 idx = v_fma(v_fma(r0+__1, __d_plus_2, c0+__1), __n_plus_2, o0);
                                                                 ^
In file included from /build/3_4_powerpc-64le-lin64/opencv/modules/core/include/opencv2/core/hal/intrin.hpp:220,
                 from /build/3_4_powerpc-64le-lin64/opencv/modules/features2d/src/sift.simd.hpp:75,
                 from /build/3_4_powerpc-64le-lin64/opencv/modules/features2d/src/sift.dispatch.cpp:76:
/build/3_4_powerpc-64le-lin64/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:1054:15: note: candidate: 'cv::hal_baseline::v_float32x4 cv::hal_baseline::v_fma(const cv::hal_baseline::v_float32x4&, const cv::hal_baseline::v_float32x4&, const cv::hal_baseline::v_float32x4&)'
 inline _Tpvec v_fma(const _Tpvec& a, const _Tpvec& b, const _Tpvec& c)      \
               ^~~~~
/build/3_4_powerpc-64le-lin64/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:1059:1: note: in expansion of macro 'OPENCV_HAL_IMPL_VSX_MULADD'
 OPENCV_HAL_IMPL_VSX_MULADD(v_float32x4)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

relates #17707

/cc @terfendail Should we have `v_fma()` for integer types?

--- a/modules/features2d/src/sift.simd.hpp
+++ b/modules/features2d/src/sift.simd.hpp
@@ -656,7 +656,7 @@ void calcSIFTDescriptor(
             v_float32 v_rco011 = v_rc01*obin, v_rco010 = v_rc01 - v_rco011;
             v_float32 v_rco001 = v_rc00*obin, v_rco000 = v_rc00 - v_rco001;
 
-            v_int32 idx = v_fma(v_fma(r0+__1, __d_plus_2, c0+__1), __n_plus_2, o0);
+            v_int32 idx = v_muladd(v_muladd(r0+__1, __d_plus_2, c0+__1), __n_plus_2, o0);
             v_store_aligned(idx_buf, idx);
 
             v_store_aligned(rco_buf,           v_rco000);

